services/enqueueing/enqueueing.go

```diff
@@ -0,0 +1,45 @@
+package enqueueing
+
+import (
+	"errors"
+
+	"google-scraper/database"
+	"google-scraper/models"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/gocraft/work"
+	"github.com/beego/beego/v2/server/web"
+)
+
+var enqueuer *work.Enqueuer
+
+func init() {
+	enqueuer = work.NewEnqueuer("google_scraper", database.GetRedisPool())
+}
+
+func EnqueueKeywordJob(keyword *models.Keyword, throttleMultiplier int64) (*work.ScheduledJob, error) {
+	if keyword.Id <= 0 {
+		return nil, errors.New("invalid keyword object")
+	}
+
+	crawlingJobName, err := web.AppConfig.String("crawlingJobName")
+	if err != nil {
+		logs.Critical("crawlingJobName is not found: ", err)
+		return nil, err
+	}
+
+	delayTimeInSeconds := getDelayTimeInSeconds(throttleMultiplier)
+	job, err := enqueuer.EnqueueIn(crawlingJobName, delayTimeInSeconds, work.Q{"keywordId": keyword.Id})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return job, nil
+}
+
+func getDelayTimeInSeconds(throttleMultiplier int64) int64 {
+	// Each job will be run two seconds later than the previous job, jobs will be enqueued immediately
+	// But will be run based on this seconds value in the future
+	return throttleMultiplier + 2
```

Comment:
Why do we need to increase the delay time from the keyword index? Can we just use the static value?

Reply:
With static value, jobs start time does delay actually. Lets say I used static value 2, the enqueueing code will be like 

```
job, err := enqueuer.EnqueueIn(crawlingJobName, 2, work.Q{"keywordId": keyword.Id})
```

Thats means, all the jobs/all keyword searching actually start run 2 second later . So here delay is not happening.

For the dynamic value, it will run like this

First keyword jobs will run immediately(secondsFromNow parameter is 0), second keyword job will run 2 seconds later(0+2 = 2), third keyword job will run 4 second later(2+2 = 4) etc.

Thats why I  need keyword index to delay the job starting time.

Reply:
I got your point now. Thanks